File: hackrank_practice/javascript-solutions/diagonal-difference.js
function diagonalDifference(arr) {
    // Write your code here
    let leftToRight = 0;
    let rightToLeft = 0;

    for (let i = 0; i < arr.length; i++) {
        leftToRight += arr[i][i];
    }
    for (let i = 0; i < arr.length; i ++) {
        rightToLeft += arr[i][(arr.length - 1) - i];
    }
    return Math.abs(leftToRight - rightToLeft);
}


// javascript version on the diagonal difference problem

File: hackrank_practice/python-solutions/diagonal-difference.py
def diagonalDifference(arr):
    left_to_right = 0
    right_to_left = 0

    for i in range(len(arr)):
        left_to_right += arr[i][i]

    for i in range(len(arr)):
        right_to_left += arr[i][(len(arr) - 1) - i]

    return abs(left_to_right - right_to_left)


# This problem wanted me to add the diagonals of a matrix
# and return the absolute value of the difference between the two.
